Documentation/advanced-configuration.md

```diff
@@ -0,0 +1,453 @@
+kind: ConfigMap
+metadata:
+  name: rook-config
+data:
+  ceph.conf: |
+    [global]
+    osd crush update on start = false
+    osd pool default size     = 2
+```
+
+Then to create or apply changes to the ConfigMap run:
+```bash
+kubectl -n rook apply -f ./rook-config.yaml
+```
+
+#### Modify The `osd` DaemonSet
+
+We will tell the OSD pods to mount our rook-config/ceph.conf as a file and add
+the `--ceph-config-override` argument pointing to that file.
+
+Start by editing the `osd` DaemonSet:
+```bash
+kubectl -n rook edit ds osd
+```
+
+You'll need to add a `volume` and `volumeMount` for the rook-config, so those
+sections might look like this to make /etc/rook/ceph.conf available in the pods:
+```yaml
+        volumeMounts:
+          - mountPath: /var/lib/rook
+            name: rook-data
+            readOnly: false
+          - mountPath: /dev
+            name: devices
+          - mountPath: /etc/rook
+            name: rook-config
+            readOnly: true
+```
+```yaml
+      volumes:
+        - name: rook-data
+          hostPath:
+            path: /var/lib/rook
+        - hostPath:
+            path: /dev
+          name: devices
+        - name: rook-config
+          configMap:
+            name: rook-config
+```
+
+Then modify the command portion to add the `--ceph-config-override` argument.
+After a little cosmetic formatting it might look like:
+```yaml
+    spec:
+      containers:
+      - command:
+          - /bin/sh
+          - -c
+          - sleep 5;
+            /usr/bin/rookd
+              osd
+              --data-dir=/var/lib/rook
+              --cluster-name=$(ROOK_CLUSTER_NAME)
+              --mon-endpoints=mon0=10.2.136.149:6790,mon1=10.2.247.0:6790,mon2=10.2.6.12:6790
+              --data-devices="sd."
+              --ceph-config-override=/etc/rook/ceph.conf
+```
+
+Lastly, restart your OSD pods by deleting them, one at a time, running `ceph -s`
+between each restart to ensure the cluster goes back to "active/clean" state.
+When that's done your new settings should be in effect.
+
+## Tweaking OSD CRUSH Settings
+
+A useful view of the [CRUSH Map](http://docs.ceph.com/docs/kraken/rados/operations/crush-map/)
+is generated with the following command:
+```bash
+ceph osd tree
+```
+
+In this section we will be tweaking some of the values seen in the output.
+
+### OSD Weight
+
+The CRUSH weight controls the ratio of data that should be distributed to each
+OSD.  This also means a higher or lower amount of disk I/O operations for an OSD
+with higher/lower weight, respectively.
+
+By default OSDs get a weight relative to their storage capacity, which maximizes
+overall cluster capacity by filling all drives at the same rate, even if drive
+sizes vary.  This should work for most use-cases, but the following situations
+could warrant weight changes:
+- Your cluster has some relatively slow OSDs or nodes. Lowering their weight can
+- reduce the impact of this bottleneck.  You're using bluestore drives
```

Comment:
the line breaks and bullet point separation isn't correct here.  You can see it better visually here: https://github.com/bzub/rook/blob/advanced-config-docs/Documentation/advanced-configuration.md#osd-weight

i think this should be 2 total bullet points, with a break in between "of this bottleneck."  and  "You're using bluestore drives"